### ISSUE TITLE
Upgrade to pdns-builder Ubuntu packages disables old bindbackend configuration

### ISSUE THREAD

**Issue by mnordhoff** [PowerDNS/pdns#7065]
- Program: Authoritative 
 - Issue type: Bug report 

### Short description

Ubuntu's packages configure the bindbackend in `/etc/powerdns/pdns.d/pdns.simplebind.conf` and `/etc/powerdns/bindbackend.conf`.

The pdns-builder packages use `/etc/powerdns/pdns.d/bind.conf` and `/etc/powerdns/named.conf`.

I was using the Ubuntu-style configuration with the previous repo.powerdns.com packages. Upgrading disabled my old configuration and set up a default new configuration. Good news: My files weren't deleted. Bad news: The new files result in 0 zones enabled.

### Environment

 - Operating system: Ubuntu 16.04
 - Software version: Upgrading from 0.0.2463gc40b447-1pdns.xenial to 0.0.15270+0.g5121e0a434-1pdns.xenial
 - Software source: PowerDNS repository

### Steps to reproduce

From a clean Ubuntu install:

1. `sudo apt install pdns-server` (4.0.0~alpha2-3build1)
2. `echo "# edit edit" | sudo tee -a /etc/powerdns/pdns.d/pdns.simplebind.conf`
3. Enable PowerDNS repository
4. `sudo apt install pdns-server=0.0.15270+0.g5121e0a434-1pdns.xenial`

Upgrading from 0.0.2463gc40b447-1pdns.xenial would be similar.

### Expected behaviour

My bindbackend configuration would continue to be used.

### Actual behaviour

My old configuration is ignored and the new default configuration is used.

### Other information

I'm not sure how I got into this situation. I don't remember if the previous repo.powerdns.com packages used the old-style configuration or the new-style configuration. Either way, you can reproduce it upgrading from Ubuntu's packages too.

```
mnordhoff@li931-94:~$ sudo apt install pdns-server=0.0.15270+0.g5121e0a434-1pdns.xenial
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libbotan-1.10-1 liblua5.3-0
Use 'sudo apt autoremove' to remove them.
The following additional packages will be installed:
  libcurl3 libluajit-5.1-2 libluajit-5.1-common libsodium18 pdns-backend-bind
The following NEW packages will be installed:
  libcurl3 libluajit-5.1-2 libluajit-5.1-common libsodium18 pdns-backend-bind
The following packages will be DOWNGRADED:
  pdns-server
0 upgraded, 5 newly installed, 1 downgraded, 0 to remove and 0 not upgraded.
Need to get 2,625 kB of archives.
After this operation, 6,066 kB of additional disk space will be used.
Do you want to continue? [Y/n]
Get:1 http://mirrors.linode.com/ubuntu xenial-updates/main amd64 libcurl3 amd64 7.47.0-1ubuntu2.9 [186 kB]
Get:2 http://mirrors.linode.com/ubuntu xenial/universe amd64 libluajit-5.1-common all 2.0.4+dfsg-1 [35.3 kB]
Get:3 http://mirrors.linode.com/ubuntu xenial/universe amd64 libluajit-5.1-2 amd64 2.0.4+dfsg-1 [205 kB]
Get:4 http://repo.powerdns.com/ubuntu xenial-auth-master/main amd64 pdns-server amd64 0.0.15270+0.g5121e0a434-1pdns.xenial [1,961 kB]
Get:5 http://mirrors.linode.com/ubuntu xenial/universe amd64 libsodium18 amd64 1.0.8-5 [144 kB]
Get:6 http://repo.powerdns.com/ubuntu xenial-auth-master/main amd64 pdns-backend-bind amd64 0.0.15270+0.g5121e0a434-1pdns.xenial [92.4 kB]
Fetched 2,625 kB in 1s (1,992 kB/s)     
Selecting previously unselected package libcurl3:amd64.
(Reading database ... 91529 files and directories currently installed.)
Preparing to unpack .../libcurl3_7.47.0-1ubuntu2.9_amd64.deb ...
Unpacking libcurl3:amd64 (7.47.0-1ubuntu2.9) ...
Selecting previously unselected package libluajit-5.1-common.
Preparing to unpack .../libluajit-5.1-common_2.0.4+dfsg-1_all.deb ...
Unpacking libluajit-5.1-common (2.0.4+dfsg-1) ...
Selecting previously unselected package libluajit-5.1-2:amd64.
Preparing to unpack .../libluajit-5.1-2_2.0.4+dfsg-1_amd64.deb ...
Unpacking libluajit-5.1-2:amd64 (2.0.4+dfsg-1) ...
Selecting previously unselected package libsodium18:amd64.
Preparing to unpack .../libsodium18_1.0.8-5_amd64.deb ...
Unpacking libsodium18:amd64 (1.0.8-5) ...
dpkg: warning: downgrading pdns-server from 4.0.0~alpha2-3build1 to 0.0.15270+0.g5121e0a434-1pdns.xenial
Preparing to unpack .../pdns-server_0.0.15270+0.g5121e0a434-1pdns.xenial_amd64.deb ...
Removing unchanged configuration file /etc/default/pdns
Removing unchanged configuration file /etc/powerdns/pdns.conf
Removing unchanged configuration file /etc/powerdns/pdns.d/pdns.local.conf
Moving configuration file /etc/powerdns/pdns.d/pdns.simplebind.conf to /etc/powerdns/pdns.d/pdns.simplebind.conf.dpkg-bak
Unpacking pdns-server (0.0.15270+0.g5121e0a434-1pdns.xenial) over (4.0.0~alpha2-3build1) ...
Selecting previously unselected package pdns-backend-bind.
Preparing to unpack .../pdns-backend-bind_0.0.15270+0.g5121e0a434-1pdns.xenial_amd64.deb ...
Unpacking pdns-backend-bind (0.0.15270+0.g5121e0a434-1pdns.xenial) ...
Processing triggers for libc-bin (2.23-0ubuntu10) ...
Processing triggers for systemd (229-4ubuntu21.4) ...
Processing triggers for ureadahead (0.100.0-19) ...
Setting up libcurl3:amd64 (7.47.0-1ubuntu2.9) ...
Setting up libluajit-5.1-common (2.0.4+dfsg-1) ...
Setting up libluajit-5.1-2:amd64 (2.0.4+dfsg-1) ...
Setting up libsodium18:amd64 (1.0.8-5) ...
Setting up pdns-server (0.0.15270+0.g5121e0a434-1pdns.xenial) ...
Installing new version of config file /etc/init.d/pdns ...
Setting up pdns-backend-bind (0.0.15270+0.g5121e0a434-1pdns.xenial) ...
Processing triggers for libc-bin (2.23-0ubuntu10) ...
Processing triggers for systemd (229-4ubuntu21.4) ...
Processing triggers for ureadahead (0.100.0-19) ...
Processing triggers for pdns-server (0.0.15270+0.g5121e0a434-1pdns.xenial) ...
```

Ubuntu's default configuration was:

```
mnordhoff@li931-94:/etc/powerdns/pdns.d$ ll *.conf
-rw------- 1 root root 104 Mar 14  2016 pdns.local.conf
-rw-r--r-- 1 root root  56 Mar 14  2016 pdns.simplebind.conf
mnordhoff@li931-94:/etc/powerdns/pdns.d$ sudo cat pdns.local.conf
# Here come the local changes the user made, like configuration of
# the several backends that exist.

mnordhoff@li931-94:/etc/powerdns/pdns.d$ cat pdns.simplebind.conf
launch+=bind
bind-config=/etc/powerdns/bindbackend.conf
```

`/etc/powerdns/bindbackend.conf` was empty.

**Comment by mnordhoff** [comment]
Okay 0.0.2463gc40b447-1pdns.xenial used the Ubuntu-style `pdns.simplebind.conf`/`bindbackend.conf` configuration.

**Comment by mnordhoff** [comment]
On Debian:

* The distro packages use `bind.conf`/`named.conf`.

* The previous repo.powerdns.com packages use `pdns.simplebind.conf`/`bindbackend.conf`.

So upgrading from e.g. 0.0.2463gc40b447-1pdns.stretch is an issue, but a new user upgrading from the distro package to the new pdns-builder packages is safe.

**Comment by pieterlexis** [comment]
This issue is multi-fold, because of the versions Ubuntu pulled in. I think the way forward is to match whatever is in the Trusty, Xenial and Bionic and match that. We do say "expect breakage" for master packages, so I am fine with this.

**Comment by pieterlexis** [comment]
Note that I only intend to fix this for upgrades from 4.1 -> 4.2
